File: 3768-check-if-digits-are-equal-in-string-after-operations-i/check-if-digits-are-equal-in-string-after-operations-i.py
class Solution:
    def hasSameDigits(self, s: str) -> bool:
        n = len(s)
        digits=[int(char) for char in s]
        while( n >2):
            for i in range (0,n-1):
                digits[i]=(digits[i]+digits[i+1])%10
            n-=1
        return digits[0]==digits[1]

File: 3768-check-if-digits-are-equal-in-string-after-operations-i/check-if-digits-are-equal-in-string-after-operations-i.cpp
class Solution {
public:
    bool hasSameDigits(string s) {
        int n=s.size();
        while(n > 2){
            for(int i=0;i<n-1;i++){
                int a=s[i]-'0';
                int b=s[i+1]-'0';
                s[i] =((a+b)%10)+'0';
            }
            n--;
        }
        return s[0]==s[1];
    }
};